### PR TITLE
push a multi-arch image

### PR DESCRIPTION
**Pull request by BenTheElder**
This will allow the ip-masq-agent DaemonSet to work on any arch out of the box (in a follow-up PR) by using a multi-arch image (manifest list).

We're planning to use the agent for http://github.com/kubernetes-sigs/kind and nominally support ADM64, ARM64 and PPCLE64 for testing, this change will make using the agent simpler.

An example is at: `gcr.io/bentheelder-kind-dev/ip-masq-agent:v2.3.0-1-g51408b3`, produced with `make all-push REGISTRY=gcr.io/bentheelder-kind-dev`.

/cc @MrHohn

### CODE REVIEW

**Comment by BenTheElder** [comment]
This pattern is borrowed from other Kubernetes Images: https://cs.k8s.io/?q=DOCKER_CLI_EXPERIMENTAL&i=nope&files=&repos=

In this case I referenced https://github.com/kubernetes/dns/blob/master/images/dnsmasq/Makefile which has a very similar Makefile to this repo.

cc @aojea

**Comment by BenTheElder** [comment]
If this change is accepted, we can follow-up with `s/-amd64//` in `ip-masq-agent.yaml` once a new image is pushed.

File: Makefile [hunk 1]
```diff
@@ -83,6 +87,11 @@ all-container: $(addprefix container-, $(ALL_ARCH))
 
 all-push: $(addprefix push-, $(ALL_ARCH))
```
Comment (MrHohn) [inline Makefile]: Remove this line?

Comment (BenTheElder) [reply Makefile]: thanks, removed the duplicate! 😅

**Pull request merged**